src/tmpfiles/tmpfiles.c

```diff
@@ -864,40 +864,15 @@ static int path_open_parent_safe(const char *path) {
 
 static int path_set_perms(Item *i, const char *path) {
         _cleanup_close_ int fd = -1;
-        struct stat *st = NULL;
 
         assert(i);
         assert(path);
 
         fd = open(path, O_NOFOLLOW|O_CLOEXEC|O_PATH);
-        if (fd < 0) {
-                int level = LOG_ERR, r = -errno;
-
-                /* Option "e" operates only on existing objects. Do not
-                 * print errors about non-existent files or directories */
-                if (i->type == EMPTY_DIRECTORY && errno == ENOENT) {
-                        level = LOG_DEBUG;
-                        r = 0;
-                }
-
-                log_full_errno(level, errno, "Adjusting owner and mode for %s failed: %m", path);
-                return r;
-        }
-
-        if (i->type == EMPTY_DIRECTORY) {
-                struct stat stbuf;
-
-                /* FIXME: introduce fd_is_dir() helper ? */
-                if (fstat(fd, &stbuf) < 0)
-                        return log_error_errno(errno, "Failed to fstat() file %s: %m", path);
-
-                if (!S_ISDIR(stbuf.st_mode))
-                        return log_error_errno(EEXIST, "'%s' already exists and is not a directory. ", path);
-
-                st = &stbuf;
-        }
+        if (fd < 0)
+                return log_error_errno(errno, "Adjusting owner and mode for %s failed: %m", path);
```

Comment:
Shouldn't this just now be `"Failed to open \"%s\" to adjust permissions: %m"` ?

Reply:
Indeed, I'll change that (even if this message will be removed at the end).